spec/interactive.test.sh

```diff
@@ -51,3 +55,13 @@ RCFILE
 ## N-I dash status: 2
 ## N-I mksh status: 1
 
+#### interactive shell runs PROMPT_COMMAND after each command
+TEST_CASE='PROMPT_COMMAND="echo hi"'
+case $SH in
+	*bash) echo "$TEST_CASE" | $SH --noprofile --norc -i;;
+	*osh) $SH --rcfile /dev/null -i -c "$TEST_CASE";;
+	*) $SH -i -c "$TEST_CASE";;
```

Comment:
Hm also this seems to be doing a ton of stuff that's different for each shell.

Are you trying to avoid the spurious stdout from bash?

I think if doing the `test/spec-bin.sh` stuff in your environment fixes it, then we should go with that instead.  I added `test/spec.sh interactive` on Travis, and it doesn't have that problem.

For now that is sort of the "golden environment".

Reply:
Yes that should have been commented. This isn't an issue with spurious output, this is because osh and bash have different ideas of what it means to be interactive.

`bash -i -c 'PROMPT_COMMAND="echo hi"'` does not give any output, but `echo 'PROMPT_COMMAND="echo hi"'' | bash -i` does.

osh goes the other way: `echo 'PROMPT_COMMAND="echo hi"' | osh -i` prints the prompt: `osh$ osh$ ^D`

I agree not having the same test for both isn't ideal but I didn't see another way to check this.